src/main/kotlin/net/atos/zac/app/planitems/PlanItemsRESTService.kt

```diff
@@ -0,0 +1,302 @@
+    private var mailService: MailService,
+    private var configuratieService: ConfiguratieService,
+    private var mailTemplateService: MailTemplateService,
+    private var policyService: PolicyService,
+    private var suspensionZaakHelper: SuspensionZaakHelper,
+    private var restMailGegevensConverter: RESTMailGegevensConverter,
+) {
+
+    companion object {
+        private const val REDEN_OPSCHORTING = "Aanvullende informatie opgevraagd"
+        private const val REDEN_PAST_FATALE_DATUM = "Aanvullende informatie opgevraagd"
+    }
+
+    @GET
+    @Path("zaak/{uuid}/humanTaskPlanItems")
+    fun listHumanTaskPlanItems(@PathParam("uuid") zaakUUID: UUID?): List<RESTPlanItem> =
+        cmmnService.listHumanTaskPlanItems(zaakUUID).let { humanTaskPlanItems ->
+            zrcClientService.readZaak(zaakUUID).let { zaak ->
+                planItemConverter.convertPlanItems(humanTaskPlanItems, zaak)
+                    .filter { it.actief }
+                    .toList()
+            }
+        }
+
+    @GET
+    @Path("zaak/{uuid}/processTaskPlanItems")
+    fun listProcessTaskPlanItems(@PathParam("uuid") zaakUUID: UUID?): List<RESTPlanItem> =
+        cmmnService.listProcessTaskPlanItems(zaakUUID).let { processTaskPlanItems ->
+            zrcClientService.readZaak(zaakUUID).let { zaak ->
+                planItemConverter.convertPlanItems(processTaskPlanItems, zaak)
+            }
+        }
+
+    @GET
+    @Path("zaak/{uuid}/userEventListenerPlanItems")
+    fun listUserEventListenerPlanItems(@PathParam("uuid") zaakUUID: UUID?): List<RESTPlanItem> =
+        cmmnService.listUserEventListenerPlanItems(zaakUUID).let { userEventListenerPlanItems ->
+            zrcClientService.readZaak(zaakUUID).let { zaak ->
+                planItemConverter.convertPlanItems(userEventListenerPlanItems, zaak)
+            }
+        }
+
+    @GET
+    @Path("humanTaskPlanItem/{id}")
+    fun readHumanTaskPlanItem(@PathParam("id") planItemId: String): RESTPlanItem =
+        convertPlanItem(planItemId)
+
+    @GET
+    @Path("processTaskPlanItem/{id}")
+    fun readProcessTaskPlanItem(@PathParam("id") planItemId: String): RESTPlanItem =
+        convertPlanItem(planItemId)
+
+    @Suppress("NestedBlockDepth")
+    private fun convertPlanItem(planItemId: String): RESTPlanItem =
+        cmmnService.readOpenPlanItem(planItemId).let { planItemInstance ->
+            zaakVariabelenService.readZaakUUID(planItemInstance).let { zaakUUID ->
+                zaakVariabelenService.readZaaktypeUUID(planItemInstance).let { zaaktypeUUID ->
+                    zaakafhandelParameterService.readZaakafhandelParameters(zaaktypeUUID).let { zaps ->
+                        planItemConverter.convertPlanItem(planItemInstance, zaakUUID, zaps)
+                    }
+                }
+            }
+        }
+
+    @POST
+    @Path("doHumanTaskPlanItem")
+    @Suppress("LongMethod")
+    fun doHumanTaskplanItem(humanTaskData: @Valid RESTHumanTaskData) {
```

Comment:
This does not work? The `@Valid` needs to be in front of the parameter, not the type?

Reply:
Yeah. You're right. Thanks!